src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -1065,6 +1098,167 @@ protected Map<String, Object> getFirstInnerHit(final Map<String, Object> searchR
+        Map<String, ArrayList<List<Object>>> countPerFieldMap = new HashMap<>();
+        for (Object innerHit : innerHits) {
+            Map<String, Object> hits = (Map<String, Object>) innerHit;
+            for (String fieldName : fieldNames) {
+                Map<String, Object> searchHits = (Map<String, Object>) hits.get(fieldName);
+                Map<String, Object> fieldHits = (Map<String, Object>) searchHits.get("hits");
+                List<Object> internalHits = (List<Object>) fieldHits.get("hits");
+                for (Object internalHit : internalHits) {
+                    Map<String, Object> internalHit1 = (Map<String, Object>) internalHit;
+                    List<Object> sorts = (List<Object>) internalHit1.get("sort");
+                    if (countPerFieldMap.containsKey(fieldName)) {
+                        ArrayList<List<Object>> sortPerField = countPerFieldMap.get(fieldName);
+                        sortPerField.add(sorts);
+                        countPerFieldMap.put(fieldName, sortPerField);
+                    } else {
+                        countPerFieldMap.put(fieldName, new ArrayList<>(Arrays.asList(sorts)));
+                    }
+                }
+
+            }
+        }
+        return countPerFieldMap;
+    }
+
+    /**
+     * Get scores of each inner hit field
+     *
+     * @param innerHits List of inner hits
+     * @param  fieldNames list of field Names of which total count need to be retrived
+     * @return Map of fieldName to scores of inner_hit from each individual hit
+     */
+    protected Map<String, ArrayList<Double>> getInnerHitsScoresPerFieldList(List<Object> innerHits, List<String> fieldNames) {
+        Map<String, ArrayList<Double>> scoresPerFieldMap = new HashMap<>();
+        for (Object innerHit : innerHits) {
+            Map<String, Object> hits = (Map<String, Object>) innerHit;
+            for (String fieldName : fieldNames) {
+                Map<String, Object> searchHits = (Map<String, Object>) hits.get(fieldName);
+                Map<String, Object> fieldHits = (Map<String, Object>) searchHits.get("hits");
+                List<Object> innerHitsOfField = (List<Object>) fieldHits.get("hits");
+                for (Object ih : innerHitsOfField) {
+                    Map<String, Object> ihMap = (Map<String, Object>) ih;
+                    Double score = (Double) ihMap.get("_score");
+                    if (scoresPerFieldMap.containsKey(fieldName)) {
+                        ArrayList<Double> scores = scoresPerFieldMap.get(fieldName);
+                        scores.add(score);
+                        scoresPerFieldMap.put(fieldName, scores);
+                    } else {
+                        scoresPerFieldMap.put(fieldName, new ArrayList<>(Arrays.asList(score)));
+                    }
+                }
+
+            }
+        }
+        return scoresPerFieldMap;
```

Comment:
Even this and the above method can be merged

Reply:
I think total hits and count can be merged together but this is just meant for scores. So we can keep this method seperate.